src/main/java/util/events/RateLimitsUpdatedEvent.java

```diff
@@ -0,0 +1,23 @@
+package util.events;
+
+/**
+ * An event that is called when new API rate limits information is available.
+ */
+public class RateLimitsUpdatedEvent extends Event{
```

Comment:
This name sounds like this event is generated when the rate has been updated. Is that correct. If this is the event that triggers the rate update, then the name is not representative. In that case the name should indicate the cause for the update. e.g. `NewApiQuotaInfoAvailable`